src/rules/spaceWithinParensRule.ts

```diff
@@ -0,0 +1,167 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+const OPTION_SPACE_SIZE_0 = 0;
+const OPTION_SPACE_SIZE_1 = 1;
+
+interface Options {
+    size: number;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "space-within-parens",
+        description: "Enforces spaces within parentheses or disallow them.",
+        hasFix: true,
+        optionsDescription: Lint.Utils.dedent`
+            You may enforce the amount of whitespace within parentheses.
+        `,
+        options: {
+            type: "array",
+            items: [
+                {
+                    type: "number",
+                    enum: [OPTION_SPACE_SIZE_0, OPTION_SPACE_SIZE_1],
+                },
+            ],
+            minLength: 0,
+            maxLength: 4,
+        },
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_NO_SPACE = "No space within parentheses";
+    public static FAILURE_NEEDS_SPACE = "Needs space within parentheses";
+    public static FAILURE_NO_EXTRA_SPACE(count: number): string {
+        return `No more than ${count} space within parentheses allowed`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new SpaceWhitinParensWalker(sourceFile, this.ruleName, parseOptions(this.ruleArguments[0])));
+    }
+}
+
+function parseOptions(whitespaceSize: number): Options {
+    let size = 0;
+    if (typeof whitespaceSize === 'number') {
+        size = whitespaceSize;
+    } else {
+        const parsedSize = parseInt(whitespaceSize, 10);
+        if (!isNaN(parsedSize)) {
+            size = parsedSize;
+        }
+    }
+    return {
+        size,
+    };
+}
+
+class SpaceWhitinParensWalker extends Lint.AbstractWalker<Options> {
+    private openingParenRegex = /^([^\(]*\()( *)[^\n\r]*$/m;
+    private closingParenRegex = /([^\s])(\s*)\)$/;
+
+    public walk(sourceFile: ts.SourceFile) {
+        const cb = (node: ts.Node): void => {
+            this.checkNode(node);
+            ts.forEachChild(node, cb);
+        };
+        return ts.forEachChild(sourceFile, cb);
+    }
+
+    private getOpenParenToken(node: ts.Node): ts.Node | undefined {
+        return Lint.childOfKind(node, ts.SyntaxKind.OpenParenToken);
```

Comment:
if you do this for every node, you could just use `forEachToken` from `tsutils` to visit every token instead of visiting every node and finding the paren tokens.